source/slang/slang-ast-val.h

```diff
@@ -166,6 +166,17 @@ class ConstantIntVal : public IntVal
     bool _isLinkTimeValOverride() { return false; }
 };
 
+FIDDLE()
+class SpecializationConstantIntVal : public IntVal
+{
+    FIDDLE(...)
+    Val* getValue() { return getOperand(1); }
```

Comment:
I believe this just need to be a `DeclRef`, and it will be cleaner if we just reuse the existing GenericParamIntVal, and rename it to DeclRefIntVal.